### PR TITLE
Github workflow for generating and uploading javadocs

### PR DESCRIPTION
**Pull request by LesTR**

### CODE REVIEW

File: .github/workflows/functions.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  echo $1 | grep -i "SNAPSHOT" > /dev/null
+  if [ $? -eq 0 ]; then
+    echo 1
+  else
+   echo 0
+  fi
+}
+
+is_datadriven_repo() {
```
Comment (je-ik) [inline .github/workflows/functions.sh]: Can we reuse this call in the push to o2 action?

File: .github/workflows/upload-javadocs.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+set -e
+
+source $(dirname $0)/functions.sh
+
+if [ "$(is_datadriven_repo ${GITHUB_REPOSITORY})" == "1" ]; then
+   VERSION=$(proxima_version)
+   
+   mvn site -Psite
```
Comment (je-ik) [inline .github/workflows/upload-javadocs.sh]: Should site (or javadoc) aggregation be called here, or does it happen automatically?

File: .github/workflows/functions.sh (hunk 1 above)
Comment (LesTR) [reply .github/workflows/functions.sh]: Changed + added prevention to overwrite ~/.ssh/id_rsa* 🤦‍♂

File: .github/workflows/upload-javadocs.sh (hunk 2 above)
Comment (LesTR) [reply .github/workflows/upload-javadocs.sh]: Sorry but I don't get the question. `mvn site -Psite` should generate and aggregate javadocs in `target/site/apidocs` which is uploaded in next step. 

Currently I see another issue - that command (`mvn site -Psite`) failed on my ntb, no idea why. Let me try to find out why....

**Comment by LesTR** [comment]
DO NOT MERGE. `mvn site -Psite` is still broken...

File: .github/workflows/sync-o2-master.sh [hunk 3]
```diff
@@ -16,15 +16,19 @@
...
-  ssh-keygen -f ~/.ssh/id_rsa -y > ~/.ssh/id_rsa.pub
+if [ "$(is_datadriven_repo ${GITHUB_REPOSITORY})" == "1" ]; then
 
+  SSH_HOME_DIR="/tmp/.ssh"
+  mkdir -p "${SSH_HOME_DIR}"
+  echo "${O2_DEPLOY_KEY}" | base64 -d > "${SSH_HOME_DIR}/id_rsa_datadriven"
+  chmod 600 "${SSH_HOME_DIR}/id_rsa_datadriven"
+  ssh-keygen -f "${SSH_HOME_DIR}/id_rsa_datadriven" -y > "${SSH_HOME_DIR}/id_rsa_datadriven.pub"
+
+  export GIT_SSH_COMMAND="ssh -i ${SSH_HOME_DIR}/id_rsa_datadriven"
```
Comment (je-ik) [inline .github/workflows/sync-o2-master.sh]: :+1:

**Comment by je-ik** [comment]
`mvn site -Psite` seems working locally. Merging and let's see :)

**Pull request merged**